src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/CodeGenerationIntegrationTest.cs

```diff
@@ -7,20 +7,39 @@
 using System.Collections.Immutable;
 using System.Linq;
 using System.Runtime.CompilerServices;
-using Microsoft.AspNetCore.Razor.Language.Extensions;
+using Microsoft.AspNetCore.Mvc.Razor.Extensions;
 using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.CodeAnalysis;
 using Roslyn.Test.Utilities;
 
 namespace Microsoft.AspNetCore.Razor.Language.IntegrationTests;
 
-public class CodeGenerationIntegrationTest(bool designTime = false)
-    : IntegrationTestBase(layer: TestProject.Layer.Compiler)
+public class CodeGenerationIntegrationTest : IntegrationTestBase
 {
+    private readonly bool designTime;
+
+    public CodeGenerationIntegrationTest(bool designTime = false)
+        : base(layer: TestProject.Layer.Compiler)
+    {
+        this.designTime = designTime;
+        BaseCompilation = BaseCompilation.AddReferences(
+            MetadataReference.CreateFromFile(typeof(TestTagHelperDescriptors).Assembly.Location));
+    }
+
     [IntegrationTestFact]
     public void SingleLineControlFlowStatements() => RunTest();
 
     [IntegrationTestFact]
-    public void CSharp8() => RunTest();
+    public void CSharp8()
+    {
+        // C# 8 features are not available in .NET Framework without polyfills
+        // so the C# diagnostics would be different between .NET Framework and .NET Core.
+        SkipVerifyingCSharpDiagnostics = ExecutionConditionUtil.IsDesktop;
```

Comment:
Did you consider making these optional params to the `IntegrationTestFact`? So we could have e.g. `[IntegrationTestFact(SkipVerifyingCSharpDiagnostics = true)]`

Reply:
Hm, haven't considered that, sounds interesting. But it looks like it would be much more difficult to propagate these options down to the IntegrationTestBase methods where I need them. If I understand correctly, I would need to add new properties to IntegrationTestFact, then add new parameters to constructors of each test, then pass the properties as constructor arguments in IntegrationTestCase. Maybe there's a simpler way? But still, for end-user the effect would be just moving something like `SkipVerifyingCSharpDiagnostics = ExecutionConditionUtil.IsDesktop;` from the test body into the attribute which doesn't seem worth the effort here. So I think I will leave as is. But happy to change this in a follow up if I misunderstood :)